src/Persistence/PersistenceManager.php

```diff
@@ -140,14 +143,53 @@ public function flush(ObjectManager $om): void
         }
     }
 
+    /**
+     * @template T of object
+     *
+     * @param T $object
+     */
+    public function autorefresh(object $object, object $clone): void
+    {
+        $strategy = $this->strategyFor($object::class);
+        $om = $strategy->objectManagerFor($object::class);
+
+        $id = $strategy->getIdentifierValues($clone);
+
+        if ($id) {
+            try {
+                $om->refresh($object);
+
+                if ($strategy->getIdentifierValues($object)) {
+                    // no identifier values means the object no longer exists
+                    return;
+                }
+            } catch (\Throwable $e) {
```

Comment:
I really don't like to do this, but, this was the only way to have something working in all situations, without being to tight with Doctrine